spec/models/user_spec.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe '#name_or_email' do
+    let!(:user) { User.new(id: 1, email: 'alice@example.com', name: '') }
+    it 'return email. if name does not exist' do
```

Comment:
```suggestion
    it 'returns email if name does not exist' do
```

ですかね

ただ、ifとかwhenに相当する部分はcontextで表現する方がRSpecらしいです。

```ruby
context "When name does not exist" do
  it "returns email" do
```